### ISSUE TITLE
L and R keys stay pressed when let go #254

### ISSUE THREAD

**Issue by Persian4859** [yuzu-emu/yuzu#3940]
Hi. Every time I press the L or R keys once, they just don't stop. I've tried changing my controller settings, switching them from docked to undocked and experimenting in the Controls tab, I even changed my controller to a PS4 controller, but nothing is working.

This hasn't been a huge problem for me until I got Pokemon Sword for a nuzlocke. Now I can't hold a box still so I can't put any Pokemon in it or take any out.

Please, I need help.

**Comment by luziferius** [comment]
Are those two buttons on your controller analog triggers?

TL/DR: Try _releasing_ the L/R triggers when prompted to assign them in yuzu.

A bit of background: Analog axis are normally treated as ranging from -2^15 (~ -32000) at the leftmost position to 2^15 (+32000) at the rightmost, whilst being centered at 0. In contrast analog triggers stay at the minimum when unpressed (-32000) and ranges to +32000 when fully pressed. But such a trigger is indistinguishable from a regular axis (that is centered at 0 when not touched). 

When mapping controller controls (in general, by any game or emulator), the user is asked to press the axis. So you press the trigger, and it registers the value changes and maps the axis as if it were in the range (0 to 32000). Now you see, 0 is way above the idle value of -32000, so it is registered as constantly pressed.

To mitigate this, press the trigger fully, _then_ start the assignment process for the button. When yuzu tells to press the button, _release_ the trigger. That will invert the mapping behaviour and yuzu will monitor the (-32000 to 0) range of that axis. That should give the desired result.
